src/lib/charm/openstack/keystone_saml_mellon.py

```diff
@@ -16,29 +16,23 @@
 import charmhelpers.core as core
 import charmhelpers.core.host as ch_host
 import charmhelpers.core.hookenv as hookenv
-import charmhelpers.core.unitdata as unitdata
 
 import charmhelpers.contrib.openstack.templating as os_templating
-import charmhelpers.contrib.openstack.utils as os_utils
 
 import charms_openstack.charm
 import charms_openstack.adapters
 
+from charms.reactive.relations import (
```

Comment:
I guess it is a matter of taste, but I prefer this:

    import charms.reactive as reactive

And then reference what you need from the module throughout the code.

Reply:
I tend to agree with @fnordahl here; the alternative syntax provides additional context to the reader that the function `endpoint_from_flag` is not in this module.

Reply:
Done